pctiler/pctiler/main.py

```diff
@@ -88,24 +84,8 @@
 
 app.include_router(health.health_router, tags=["Liveliness/Readiness"])
 
-app.add_middleware(RequestTracingMiddleware, service_name=ServiceName.TILER)
-
-
-@app.middleware("http")
-async def _timeout_middleware(
-    request: Request, call_next: Callable[[Request], Awaitable[Response]]
-) -> Response:
-    """Add a timeout to all requests."""
-    return await timeout_middleware(request, call_next, timeout=settings.request_timout)
-
-
-@app.middleware("http")
-async def _handle_exceptions(
-    request: Request, call_next: Callable[[Request], Awaitable[Response]]
-) -> Response:
-    return await handle_exceptions(request, call_next)
-
-
+app.add_exception_handler(Exception, http_exception_handler)
```

Comment:
Can you confirm that if you make an invalid POST request (like bad data) we still get the nice error message? I couldn't quickly confirm what the behavior of `app.add_exception_handler` is with Exception subclasses / order. I see that we reraise `HTTPException` in our handler, and just want to make sure FastAPI's still handles it.

Reply:
![image](https://github.com/microsoft/planetary-computer-apis/assets/100631414/8cee5f83-2182-48b4-a2f2-9511edeb9285)
okie so this is an instance of internal server error

Reply:
![image](https://github.com/microsoft/planetary-computer-apis/assets/100631414/80f3e9b3-a6dc-480b-bf9a-0e0ec77e525e)
this is 404 collection not found